examples/drivenet/drivenet_close_loop.ipynb

```diff
@@ -0,0 +1,326 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# DriveNet Close Loop Evaluation\n",
+    "\n",
+    "**Note: this notebook assumes you've already run the [training notebook](./drivenet_train.ipynb) and stored your model successfully.**\n",
+    "\n",
+    "## What is a close loop evaluation?\n",
+    "In close loop evaluation DriveNet is in **full control of the AV**. At each time step, we predict the future trajectory and then we move the AV int he first of the DriveNet's predictions. \n",
+    "\n",
+    "At Lyft, we refer to this process with the term **unroll**\n",
+    "\n",
+    "\n",
+    "## What is a good close loop metric?\n",
+    "For this setting metrics are particularly challenging. In fact, we would like to penalise some of the drifting (e.g. going off road or in the opposite lane) while at the same time allow others (e.g. different speed profiles)\n",
+    "\n",
+    "At Lyft L5, we use a substantial sets of different metrics to capture dangerous manoeuvres and behaviours. \n",
+    "\n",
+    "For the sake of simplicity, in this notebook we will be using a very simple proxy to detect if our model is driving in a sensible way, composed of two different metrics:\n",
+    "\n",
+    "### Collisions\n",
+    "Our AV should avoid collisions with other agents. This sounds trivial, but it's actually more complex than how it looks. In fact, while our AV is fully controlled by Drivenet, other agents are not (a setting we call **log replay**).\n",
+    "\n",
+    "If our AV was slower than the recorded one, chasing agents might bump into us. Clearly, this won't happen in a real setting where other agents can react to our behaviours.\n",
+    "\n",
+    "Still, this metric is useful to assess how much our DriveNet follows the rules of the road.\n",
+    "\n",
+    "However, if we only considered collision, we might under penalise some potentially dangerous situations like driving off-road.\n",
```

Comment:
I don't like how this paragraph is structured: 'not as trivial as it sounds' -> this makes me more think of is detecting collisions trivial? Also, bad rear collisions are more caused by log-replay and not because this metric is bad ... ping me if unclear.

Reply:
Under penalised: it's not penalized at all then - would also reformulate this.

Reply:
collisionS